dokka-runners/runner-gradle-plugin-classic/build.gradle.kts

```diff
@@ -1,71 +0,0 @@
-/*
- * Copyright 2014-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
- */
-
-import dokkabuild.PublicationName
-import dokkabuild.overridePublicationArtifactId
-
-plugins {
-    id("dokkabuild.gradle-plugin")
-    id("dokkabuild.dev-maven-publish")
-    alias(libs.plugins.kotlinx.binaryCompatibilityValidator)
-}
-
-overridePublicationArtifactId("dokka-gradle-plugin", PublicationName.GRADLE_PLUGIN)
-
-dependencies {
-    // the version is required for Gradle plugin publishing
-    api("org.jetbrains.dokka:dokka-core:$version")
-
-    compileOnly(libs.gradlePlugin.kotlin)
-    compileOnly(libs.gradlePlugin.kotlin.klibCommonizerApi)
-    compileOnly(libs.gradlePlugin.android)
-
-    testImplementation(kotlin("test"))
-    testImplementation(libs.gradlePlugin.kotlin)
-    testImplementation(libs.gradlePlugin.kotlin.klibCommonizerApi)
-    testImplementation(libs.gradlePlugin.android)
-    testImplementation("org.jetbrains.dokka:dokka-test-api:$version")
-}
-
-gradlePlugin {
-    plugins {
-        create("dokka") {
-            id = "org.jetbrains.dokka"
-            implementationClass = "org.jetbrains.dokka.gradle.DokkaPlugin"
-
-            displayName = "Dokka plugin"
-            description = "Dokka is an API documentation engine for Kotlin"
-            @Suppress("UnstableApiUsage")
-            tags.addAll("dokka", "kotlin", "kdoc", "android", "documentation", "api")
-        }
-    }
-}
-
-// Gradle will put its own version of the stdlib in the classpath, so not pull our own we will end up with
-// warnings like 'Runtime JAR files in the classpath should have the same version'
-listOf(
-    configurations.api,
-    configurations.implementation,
-    configurations.runtimeOnly
-).forEach {
-    it.configure { excludeGradleCommonDependencies() }
-}
-
-/**
- * These dependencies will be provided by Gradle, and we should prevent version conflict
- * Code taken from the Kotlin Gradle plugin:
- * https://github.com/JetBrains/kotlin/blob/70e15b281cb43379068facb82b8e4bcb897a3c4f/buildSrc/src/main/kotlin/GradleCommon.kt#L72
- */
-fun Configuration.excludeGradleCommonDependencies() {
```

Comment:
The build config has been reworked so that the dependencies aren't added, so this exclusion does nothing.

If you run `./gradlew :dokka-gradle-plugin:publishDokkaPluginMarkerMavenPublicationToDevMavenRepo` and check `dokka-runners/dokka-gradle-plugin/build/dev-maven-repo/org/jetbrains/dokka/dokka-gradle-plugin/2.0.20-SNAPSHOT/dokka-gradle-plugin-2.0.20-SNAPSHOT.pom`, then there's no extra files.

We should make a test for this though... I've created a task.

https://youtrack.jetbrains.com/issue/KT-70503/Add-test-to-verify-Dokka-Plugin-does-not-expose-conflicting-dependencies

Reply:
serialization brings stdlib as a transitive dependency, that is why we need this code block

Reply:
I tried hard to try and discover what the issue is, but I wasn't able to discover what could go wrong. Whenever Dokka was added as a buildSrc dependency, the kotlin-stdlib and other dependencies were correctly constrained to the embedded Gradle version.

When this exclusions config was added in https://github.com/Kotlin/dokka/pull/2570, at the time DGP used `kotlin("jvm")` to build the project. This is probably what caused the issue, and the exclusions helped with that. (KGP is also built with `kotlin("jvm")`.) DGP no longer uses `kotlin("jvm")`, so I don't believe it's necessary. However, out of an abundance of caution, I've added the exclusions.